### PR TITLE
Bash script added for shellcheck verification

### LINKED ISSUE #482: Add script to run shellcheck on bash scripts in the repository
/kind feature

**Describe the solution you'd like**
A lot of CI jobs depend on our bash scripts, we should add shellcheck as part of presubmit to avoid small mistakes.

- Add a bash script to run shellcheck on all the scripts in the repository. Derived from [Kubernetes Shellcheck](https://github.com/kubernetes/kubernetes/blob/master/hack/verify-shellcheck.sh)
- Add `verify-shellcheck` make target and include it as a part of `make verify`.
- Fix shellcheck warnings in the repository.

### PR DESCRIPTION
**Pull request by sayantani11**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR adds a bash script for shellcheck verification, and makes the changes required for the feature.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #482 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Shellcheck verfication script added
```

### CODE REVIEW

**Review by mehabhalodiya — COMMENTED** [review]
LGTM 🎉

File: hack/verify-shellcheck.sh [hunk 1]
```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+#Copyright 2018 The Kubernetes Authors.
```
Comment (cpanato) [inline hack/verify-shellcheck.sh:2]:
```suggestion
# Copyright 2022 The Kubernetes Authors.
```

**Review by cpanato — CHANGES_REQUESTED** [review]
thanks for that
a small request

**Review by cpanato — APPROVED** [review]
thanks!
/approve
/lgtm

**Pull request merged**